Add client authentication support for ticdc (#17227)
pingcap/docs: ticdc/ticdc-client-authentication.md

---
title: TiCDC Client Authentication
summary: Introduce how to perform TiCDC client authentication using the command-line tool or OpenAPI.
---

# TiCDC Client Authentication

> Starting from v8.1.0, TiCDC supports client authentication using Mutual Transport Layer Security (mTLS) or TiDB username and password. 

Reviewer: ```suggestion
Since v8.1.0, TiCDC supports client authentication using Mutual Transport Layer Security (mTLS) or TiDB username and password. 
```

- mTLS authentication provides security control at the transport layer, enabling TiCDC to verify the client identity.
- TiDB username and password authentication provides security control at the application layer, ensuring that only authorized users can log in through the TiCDC node.

These two authentication methods can be used either independently or in combination to meet different scenarios and security requirements. 

> **Note:**
>

> > TiCDC client authentication is only available when [TLS is enabled](/enable-tls-between-clients-and-servers.md).

Author: ```suggestion
> In the production environment, it is recommended to use TiCDC client authentication only when [TLS is enabled](/enable-tls-between-clients-and-servers.md).
```

> > In the production environment, it is recommended to use TiCDC client authentication only when [TLS is enabled](/enable-tls-between-clients-and-servers.md).

Reviewer 2: ```suggestion
> In production environments, it is recommended to use TiCDC client authentication only when [TLS is enabled](/enable-tls-between-clients-and-servers.md).
```

> > In the production environment, it is recommended to use TiCDC client authentication only when [TLS is enabled](/enable-tls-between-clients-and-servers.md).

Reviewer 2: 中文这里又更新了😂

为了保证网络访问的安全性，强烈建议仅在[开启 TLS 加密传输](/enable-tls-between-clients-and-servers.md)的情况下，使用 TiCDC 客户端鉴权功能。如果不开启 TLS 加密传输，用户名和密码将会以明文的方式通过网络传输，这会带来严重的泄露风险。

> > In the production environment, it is recommended to use TiCDC client authentication only when [TLS is enabled](/enable-tls-between-clients-and-servers.md).

Author: ```suggestion
> To ensure the security of network access, it is strongly recommended to use TiCDC client authentication only when [TLS is enabled](/enable-tls-between-clients-and-servers.md). If TLS is not enabled, the username and password are transmitted as plaintext over the network, which can lead to serious credential leaks.
```

## Use mTLS for client authentication

1. In the TiCDC server, configure the `security.mtls` parameter as `true` to enable mTLS authentication:

    ```toml
    [security]
    # This parameter controls whether to enable the TLS client authentication. The default value is false.
    mtls = true
    ```

> 2. Configure the client certificate:

Reviewer 2: ```suggestion
2. Configure the client certificate.
```

    ```bash
    curl -X GET http://127.0.0.1:8300/api/v2/status --cert client.crt --key client.key --cacert ca.crt
    ```

    </div>
    </SimpleTab>

## Use TiDB user for client authentication

1. [Create a user](/sql-statements/sql-statement-create-user.md) in TiDB and grant the user permission to log in from the TiCDC node.

    ```sql
    CREATE USER 'test'@'ticdc_ip_address' IDENTIFIED BY 'password';
    ```

2. In the TiCDC server, configure `security.client-user-required` and `security.client-allowed-user` to enable username and password authentication:

    ```toml
    [security]
    # This parameter controls whether to use username and password for client authentication. The default value is false.
    client-user-required = true
    # This parameter lists the usernames that are allowed for client authentication. Authentication requests with usernames not in this list will be rejected. The default value is null.
    client-allowed-user = ["test"]
    ```

> 3. Specify the username and password of the user created in step 1:

Reviewer 2: ```suggestion
3. Specify the username and password of the user created in step 1.
```
